auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailActivity.java

```diff
@@ -138,15 +139,22 @@ public void onClick(View view) {
     private void registerUser(String email, final String name, final String password) {
         final FirebaseAuth firebaseAuth = mActivityHelper.getFirebaseAuth();
         // create the user
-        firebaseAuth.createUserWithEmailAndPassword(email, password)
-                .addOnFailureListener(new TaskFailureLogger(TAG, "Error creating user"))
+        FirebaseUser user = firebaseAuth.getCurrentUser();
+        Task<AuthResult> task;
+        setIntent(getIntent().putExtras(mActivityHelper.getMergeFailedIntent()));
```

Comment:
Woops! That's actually not supposed to happen at all since we're linking with a new credential (eg the account doesn't exist yet).